src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java

```diff
@@ -94,14 +94,23 @@ public CourseView showCourseView(@PathVariable long workgroupId, @PathVariable l
 	@RequestMapping(value = "/api/courseView/sectionGroups", method = RequestMethod.POST, produces="application/json")
 	@ResponseBody
 	public SectionGroup createSectionGroup(@RequestBody SectionGroup sectionGroup, HttpServletResponse httpResponse) {
-		if (sectionGroup.getCourse() == null) return null;
+		if (sectionGroup == null) {
+			httpResponse.setStatus(HttpStatus.NOT_FOUND.value());
+			return null;
+		}
 
-		// TODO: Consider how we can improve the authorizer
 		Course course = courseService.getOneById(sectionGroup.getCourse().getId());
```

Comment:
Course = getCourse().getId()?

Reply:
Line 102 is verifying that the course id that comes from the user (nested inside the sectionGroup param) does reference a persisted course. It 'should' exist, but entity Ids from the user should not be trusted.

Reply:
Oops, my bad.